tests/transactions_example.rs

```diff
@@ -0,0 +1,91 @@
+#![cfg(all(feature = "tokio-runtime", not(feature = "tokio-sync")))]
+
+// START TRANSACTIONS EXAMPLE
+use mongodb::{
+    bson::{doc, Document},
+    error::{Result, TRANSIENT_TRANSACTION_ERROR, UNKNOWN_TRANSACTION_COMMIT_RESULT},
+    options::{Acknowledgment, ReadConcern, TransactionOptions, WriteConcern},
+    Client,
+    ClientSession,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
```

Comment:
The previous example didn't include a `main` and it doesn't look like other languages' do either - maybe move this outside the `// START` comment?

Reply:
The pseudocode this example is based off of (transactions retry example 3 in DRIVERS-488) does include construction of the client/session and some of the scripting languages' examples (e.g. JS and Python) show those steps. I think it's helpful to demonstrate how to start a session and emphasize that `run_transaction_with_retry` is the entry point here.